src/libraries/Microsoft.PowerFx.Core/Public/Values/RecordValue.cs

```diff
@@ -15,21 +16,21 @@ namespace Microsoft.PowerFx.Types
     /// </summary>
     public abstract class RecordValue : ValidFormulaValue
     {
+#pragma warning disable CA1721 // CA1721: Property names should not match get methods
         /// <summary>
         /// Fields and their values directly available on this record. 
         /// The field names should match the names on <see cref="Type"/>. 
         /// </summary>
-        public IEnumerable<NamedValue> Fields
-        {
-            get
+        public IEnumerable<NamedValue> Fields => GetFields(CancellationToken.None);
+
+        public IEnumerable<NamedValue> GetFields(CancellationToken cancellationToken)
```

Comment:
>GetFields [](http://example.com/codeflow?start=39&length=9)

Do we need it here?

Reply:
Same answer, this is in link with Dataverse where Fields enumeration needs a CancellationToken.
We could also make GetFieldsAsync and return a Task<>